File: starter-bots/KotlinBot/src/main/kotlin/arctica/service/model/dto/ResourceNode.kt
package arctica.service.model.dto

import arctica.service.model.GameObject
import arctica.service.model.Position
import java.util.*

class ResourceNode(
    val type: Int,
    val amount: Int,
    val maxUnits: Int,
    val currentUnits: Int,
    val reward: Int,
    val workTime: Int,
    val regenerationRate: RegenerationRate,
    id: UUID,
    gameObjectType: Int,
    position: Position

) : GameObject(id, gameObjectType, position)

data class RegenerationRate(
    val ticks: Int,
    val amount: Int
)

File: starter-bots/PythonBot/ResourceNode.py
from GameObject import GameObject

class ResourceNode(GameObject):
    def __init__(self, guid, game_object_type, position, resource_type, amount, max_units, current_units, reward, work_time) -> None:
        super().__init__(guid, game_object_type, position)
        self.resource_type = resource_type
        self.amount = amount
        self.max_units = max_units
        self.current_units = current_units
        self.reward = reward
        self.work_time = work_time